com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -852,6 +853,26 @@ private void OnEnable()
             }
         }
 
+        private void Awake()
+        {
+            UnityEngine.SceneManagement.SceneManager.sceneUnloaded += OnSceneUnloaded;
+        }
+
+        // Ensures that the NetworkManager is cleaned up before OnDestroy is run on NetworkObjects and NetworkBehaviours when unloading a scene with a NetworkManager
+        private void OnSceneUnloaded(Scene scene)
+        {
+            if (scene == gameObject.scene)
+            {
+                OnDestroy();
+            }
+        }
```

Comment:
so, we're saying if the scene of NetworkManager's scene is being unloaded (therefore it'll also be unloaded as a part of it), we'd manually call OnDestroy() method. wouldn't it be called anyway as a part of scene unloading process? is this block needed?

Reply:
It is needed because we want `OnDestroy` on the NetworkManager to be called before `OnDestroy` gets called on the NetworkObjects in the scene. And this seems to be the only way we can achieve that. (ScriptExecutionOrder does not apply to OnDestroy)

Reply:
oh, now I get it. well, sounds like a band-aid to me :/ but let's get that band-aid until we have a better solution.

Reply:
This won't fix the issues with NetworkObjects being destroyed on the client side (when pooled) or if you are transitioning between single loading mode scenes when the NetworkManager is within the DontDestroyOnLoad scene because that gets unloaded last (after all other scenes are unloaded).

Reply:
Yes that's correct, we are not fixing those issues here. This only solves application quit and unloading a non DontDestroyOnLoad NetworkManager from a scene.